addon/components/impagination-collection.js

```diff
@@ -44,11 +38,15 @@ export default Ember.Component.extend({
     this.set('datasetState', this.get('dataset.state'));
   }),
 
+  safeSet: function(key, value) {
+    if (!this.isDestroyed) { this.set(key, value); }
+  },
```

Comment:
This is interesting. I like it. I'm assuming this came out of test errors saying something like "trying to call set on a destroyed object"? I've struggled with that a ton too.